client/src/main/java/connection/Client.java

```diff
@@ -1,35 +1,64 @@
 package connection;
 
 import java.io.*;
-import java.net.Socket;
+import java.util.Scanner;
 
 public class Client {
 
-    Socket socket;
-    BufferedWriter writer;
-    BufferedReader reader;
+    static final String CHARSET = "UTF-8";
+
+    Connector connector;
+    Sender out;
+    Receiver in;
 
     public void setup(String host, String port) {
         try {
-            socket = new Socket(host, Integer.valueOf(port));
-            writer = new BufferedWriter(new OutputStreamWriter(socket.getOutputStream(), "UTF-8"));
-            reader = new BufferedReader(new InputStreamReader(socket.getInputStream(), "UTF-8"));
+            connector = SocketConnector.from(host, Integer.valueOf(port));
+            in = MessageIn.from(connector);
+            out = MessageOut.from(connector);
         } catch (IOException e) {
             e.printStackTrace();
```

Comment:
add logger in the future. Only printStackTarce no action needed? If after setup other function will use connector, in or out will not know which of them is invalid